src/GrafanaDashboardDeploy/src/GrafanaClient.cs

```diff
@@ -0,0 +1,327 @@
+                        _logger.LogWarning("Status {}: {}", response.StatusCode, response.ReasonPhrase);
+                        _logger.LogDebug(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
+
+                        // TODO: How to handle error?
+                    }
+
+                    var st = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    using (var sr = new StreamReader(st))
+                    using (var jr = new JsonTextReader(sr))
+                    {
+                        folderResponse = await JObject.LoadAsync(jr).ConfigureAwait(false);
+                    }
+                }
+            }
+
+            return folderResponse;
+        }
+
+        public async Task<JObject> CreateDatasourceAsync(JObject datasource)
+        {
+            var uri = new Uri(new Uri(BaseUrl), "/api/datasources");
+            JObject responseJson;
+
+            var stream = new MemoryStream();
+            using (var textWriter = new StreamWriter(stream))
+            using (var jsonStream = new JsonTextWriter(textWriter))
+            {
+                var serializer = new JsonSerializer();
+                serializer.Serialize(jsonStream, datasource);
+
+                jsonStream.Flush();
+                stream.Position = 0;
+
+                using (var content = new StreamContent(stream))
+                {
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                    HttpResponseMessage response = await _client.PostAsync(uri, content).ConfigureAwait(false);
+
+                    if (!response.IsSuccessStatusCode)
+                    {
+                        _logger.LogWarning("Response does not indicate success");
+                        _logger.LogWarning("Status {}: {}", response.StatusCode, response.ReasonPhrase);
+                        _logger.LogDebug(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
+
+                        // TODO: How to handle error?
+                    }
+
+                    var st = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    using (var sr = new StreamReader(st))
+                    using (var jr = new JsonTextReader(sr))
+                    {
+                        responseJson = await JObject.LoadAsync(jr).ConfigureAwait(false);
+                    }
+                }
+            }
+
+            return responseJson;
+        }
+
+        public async Task<JObject> CreateDashboardAsync(JObject dashboard, int folderId)
+        {
+            var uri = new Uri(new Uri(BaseUrl), "/api/dashboards/db");
+
+            var dashboardBody = new JObject(
+                new JProperty("dashboard", dashboard),
+                new JProperty("folderId", folderId),
+                new JProperty("overwrite", false));
+
+            JObject responseJson;
+
+            var stream = new MemoryStream();
```

Comment:
This should be using'd too.

Reply:
(most of these comments apply to the other method too. To the point where you might want to factor out the "send a JObject to grafana, get one back, and log the error if there is one)

Reply:
`stream` is disposed when the `StreamContent()` that wraps it below is disposed. Visual Studio catches this, which confused me until I did some reading. 

But yes, I'll look at refactoring this.

Reply:
In general, it's safer to dispose things you create, even if someone else might dispose them too. Stuff get refactored, behaviors change, and it's always safe to double dispose.  Ownership is a weirdly difficult thing in C#.

Technically the StreamWriter also disposes it (because we didn't pass "false" in the constructor), and I'd bet the JsonTextWriter does too (by disposing the StreamWriter).  This object is almost certainly already disposed 3 times.  But a 4th one won't hurt anything, and it might prevent a weird bug in the future if we refactor out some of this.

Reply:
Sounds like a good way to code defensively. I'll change it.